contractcourt/anchor_resolver.go

```diff
@@ -195,7 +202,8 @@ func (c *anchorResolver) IsResolved() bool {
 // state required for the proper resolution of a contract.
 //
 // NOTE: Part of the ContractResolver interface.
-func (c *anchorResolver) SupplementState(_ *channeldb.OpenChannel) {
+func (c *anchorResolver) SupplementState(state *channeldb.OpenChannel) {
+	c.chanType = state.ChanType
```

Comment:
When creating the arbitrator [here](https://github.com/lightningnetwork/lnd/blob/master/contractcourt/chain_arbitrator.go#L356), the channel type can be accessed via `ResolverConfig.ChannelArbitratorConfig.Channel` if we extend `ArbChannel` interface. Unless the channel type can be changed dynamically, `SupplementState` seems to be a bit hacky.

Reply:
So we don't always have access to the channel field there. When we make arbs for closing channels `ChainArbitrator.Start()` we've lost access to that struct. 

`SupplementState` was originally added to avoid adding more state to the resolvers, when it could be obtained elsewhere at runtime.